tests/SprykerTest/Glue/AppMerchantBackendApi/Plugin/MerchantConfirmDisconnectionRequestValidatorPluginTest.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ * This file is part of the Spryker Suite.
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerTest\Glue\AppMerchantBackendApi\Plugin;
+
+use Codeception\Test\Unit;
+use Generated\Shared\Transfer\GlueErrorConfirmTransfer;
+use Generated\Shared\Transfer\GlueRequestTransfer;
+use Generated\Shared\Transfer\MerchantTransfer;
+use Spryker\Glue\AppMerchantBackendApi\AppMerchantBackendApiConfig;
+use Spryker\Glue\AppMerchantBackendApi\AppMerchantBackendApiDependencyProvider;
+use Spryker\Glue\AppMerchantBackendApi\Dependency\Facade\AppMerchantBackendApiToTranslatorFacadeInterface;
+use Spryker\Glue\AppMerchantBackendApi\Plugin\GlueApplication\MerchantConfirmDisconnectionRequestValidatorPlugin;
+use Spryker\Glue\AppPaymentBackendApi\Mapper\Payment\GlueRequestPaymentMapper;
+use SprykerTest\Glue\AppMerchantBackendApi\AppMerchantBackendApiPluginTester;
+
+/**
+ * Auto-generated group annotations
+ *
+ * @group SprykerTest
+ * @group Glue
+ * @group AppMerchantBackendApi
+ * @group Plugin
+ * @group MerchantConfirmDisconnectionRequestValidatorPluginTest
+ * Add your own group annotations below this line
+ */
+class MerchantConfirmDisconnectionRequestValidatorPluginTest extends Unit
+{
+    /**
+     * @var string
+     */
+    protected const TENANT_IDENTIFIER = 'tenant-identifier';
+
+    /**
+     * @see \Spryker\Glue\AppKernel\Plugin\GlueApplication\AbstractConfirmDisconnectionRequestValidatorPlugin::HEADER_CONFIRMATION_STATUS
+     *
+     * @var string
+     */
+    protected const HEADER_CONFIRMATION_STATUS = 'x-confirmation-status';
+
+    protected AppMerchantBackendApiPluginTester $tester;
+
+    public function testMerchantConfirmDisconnectionRequestValidatorPluginReturnsErrorIfThereIsNoTenantIdentifierGivenInTheRequest(): void
+    {
+        // Arrange
+        $this->tester->setDependency(AppMerchantBackendApiDependencyProvider::FACADE_TRANSLATOR, $this->getTranslatorFacadeMock());
+
+        $merchantConfirmDisconnectionRequestValidatorPlugin = new MerchantConfirmDisconnectionRequestValidatorPlugin();
+
+        // Act
+        $glueRequestValidationTransfer = $merchantConfirmDisconnectionRequestValidatorPlugin
+            ->validate(new GlueRequestTransfer());
+
+        // Assert
+        $this->assertFalse($glueRequestValidationTransfer->getIsValid());
+        $this->assertCount(1, $glueRequestValidationTransfer->getErrors());
+        $this->assertSame(
+            AppMerchantBackendApiConfig::ERROR_CODE_PAYMENT_DISCONNECTION_TENANT_IDENTIFIER_MISSING,
+            $glueRequestValidationTransfer->getErrors()[0]->getCode(),
+        );
+    }
+
+    public function testMerchantConfirmDisconnectionRequestValidatorPluginReturnsSuccessIfThereAreNoMerchantsForTenantIdentifierGivenInTheRequest(): void
```

Comment:
Merchants are not in the request but in the database. Check all other method as well, please.

Reply:
Rephrased.